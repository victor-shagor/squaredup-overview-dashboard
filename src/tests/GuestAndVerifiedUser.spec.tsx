import { render, fireEvent, screen } from "@testing-library/react";
import GuestAndVerifiedUsers from "components/GuestsAndVerifiedUsers/GuestAndVerifiedUser";
import renderer from "react-test-renderer";

const mockData = {
  guestUser: {
    name: "Guest Users",
    data: [225, 1419, 1665, 194, 3, 3, 3, 2],
  },
  verifiedUser: {
    name: "Verified Users",
    data: [7, 329, 451, 643, 310, 559, 567, 115],
  },
  category: ["2022", "2021", "2020", "2019", "2018", "2017", "2016", "2015"],
};
jest.mock("react-apexcharts", () => {
  return {
    __esModule: true,
    default: () => {
      return <div />;
    },
  };
});

describe("guest and verified user component", () => {
  it("render active user component with the correct style", () => {
    const onClickToggle = jest.fn();
    render(
      <GuestAndVerifiedUsers
        columnChartToggle={"yearly"}
        guestAndVerifiedUsers={mockData}
        onClickToggle={onClickToggle}
      />
    );
    expect(screen.getByTestId("user-metric-container")).toHaveStyle(
      `height:25rem`
    );
    expect(screen.getByTestId("user-metric-container")).toHaveStyle(
      `width:96.5%`
    );
  });

  it("should change active toggle button when  clicked", () => {
    const onClickToggle = jest.fn();
    render(
      <GuestAndVerifiedUsers
        columnChartToggle={"yearly"}
        guestAndVerifiedUsers={mockData}
        onClickToggle={onClickToggle}
      />
    );
    const dailyToggleButton = screen.getByTestId("daily-toggle-button");
    const monthlyToggleButton = screen.getByTestId("monthly-toggle-button");
    const yearlyToggleButton = screen.getByTestId("yearly-toggle-button");

    expect(yearlyToggleButton).toHaveStyle(`background-color: #2C2C2C`);
    expect(monthlyToggleButton).toHaveStyle(`background-color: white`);
    expect(dailyToggleButton).toHaveStyle(`background-color: white`);
    5;
    fireEvent.click(monthlyToggleButton);

    expect(onClickToggle).toHaveBeenCalled();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <GuestAndVerifiedUsers
          columnChartToggle={"yearly"}
          guestAndVerifiedUsers={mockData}
          onClickToggle={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
