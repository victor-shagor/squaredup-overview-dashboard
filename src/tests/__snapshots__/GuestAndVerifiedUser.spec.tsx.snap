// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-gsnTZi fNokON activechart"
  data-testid="user-metric-container"
  height="25rem"
  width="96.5%"
>
  <div
    className="details"
  >
    <p>
      Guest and Verified Users
    </p>
    <div
      className="columnChartToggleDiv"
    >
      <div
        className="columnChartToggle"
      >
        <button
          className="activeToggleButton"
          data-testid="yearly-toggle-button"
          onClick={[Function]}
        >
          Yearly
        </button>
        <button
          className="toggleButton"
          data-testid="monthly-toggle-button"
          onClick={[Function]}
        >
          Monthly
        </button>
        <button
          className="toggleButton"
          data-testid="daily-toggle-button"
          onClick={[Function]}
        >
          Daily
        </button>
      </div>
    </div>
  </div>
  <div
    className="chart"
  >
    <div />
  </div>
</div>
`;
